torchrl/modules/models/multiagent.py

```diff
@@ -125,6 +125,21 @@ def forward(self, *inputs: Tuple[torch.Tensor]) -> torch.Tensor:
 
         return output
 
+    def reset_parameters(self):
+        def vmap_reset_module(module, *args, **kwargs):
+            def reset_module(params):
+                with params.to_module(module):
+                    module.reset_parameters()
+                    return params
+
+            return torch.vmap(reset_module, *args, **kwargs)
+
+        if not self.share_params:
+            vmap_reset_module(self._empty_net, randomness="different")(self.params)
```

Comment:
I would like to discuss the vmap randomness.

```python
    @property
    def _vmap_randomness(self):
        if self.initialized:
            return "error"
        return "same"
```

Why would this be a class property and why are we having those values?

For me this should be

```python
    @property
    def _vmap_randomness(self):
        return "different"
```

Reply:
In this case it needs to be different to have a different reset values for each agent.

But also in the forward pass I feel like it should be "different"

Reply:
IMO it should be
```python
    @property
    def _vmap_randomness(self):
        if self.initialized:
            return self.vmap_randomness
        return "different"
```

and users are in charge of telling the module what randomness they want.

Reply:
Could you explain this a bit? Why do we have a switch on initialization?

Reply:
For init you need "different" because you must have different weights for each net.
But in other settings you can't tell, and the best is to let the user choose. 
They may as well want the same random number for each element of the batch

Reply:
if I try to change this to 
```python
    @property
    def _vmap_randomness(self):
        if self.initialized:
            return self.vmap_randomness
        return "different"
```
the lazy layers will crash

Reply:
> the lazy layers will crash

this is a statement that is hard to reproduce, can you share more?

For instance this works fine on my end:
```python
from torchrl.modules import MLP
from tensordict import TensorDict
import torch
from functorch import dim

d0 = dim.dims(1)
modules = [torch.nn.Linear(2, 3) for _ in range(3)]

td = TensorDict.from_modules(*modules, as_module=True)

def reset(td):
    with td.to_module(modules[0]):
        modules[0].reset_parameters()
    return td

td = torch.vmap(reset, randomness="same")(td)
print(td["weight"])
td = torch.vmap(reset, randomness="different")(td)
print(td["weight"])
```
the first produces a stack of identical tensors, the second different